applications/ShapeOptimizationApplication/tests/shape_optimization_test_factory.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Import KratosUnittest
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+import KratosMultiphysics.kratos_utilities as kratos_utils
+
+# Other imports
+import os
+
+class controlledExecutionScope:
+    def __init__(self, scope):
+        self.currentPath = os.getcwd()
+        self.scope = scope
+
+    def __enter__(self):
+        os.chdir(self.scope)
+
+    def __exit__(self, type, value, traceback):
+        os.chdir(self.currentPath)
+
+# ==============================================================================
+class ShapeOptimizationTestFactory(KratosUnittest.TestCase):
+    # --------------------------------------------------------------------------
+    def setUp(self):
+        pass
+
+    # --------------------------------------------------------------------------
+    def test_execution(self):
+        with controlledExecutionScope(os.path.dirname(os.path.realpath(__file__))+"/"+self.execution_directory):
+            test_status = os.system("runkratos "+self.execution_file+" >>run_test.log")
```

Comment:
@roigcarlo do you know a better way of doing this?

I think there was a problem with running the script using `import`

Reply:
then i suggest to use ```os.path-join()```

Reply:
That should not be used under any circumstance. If you **really** need to execute a separate process use the subprocess module and redirect the output:

```Python
process = subprocess.Popen([command, script, options], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

for the command you may want to use this instead of just `runkratos`:

```Python
os.path.dirname(GetModulePath('KratosMultiphysics'))+'/'+'runkratos'
```

and if you want to check the output:

```Python
process_stdout, process_stderr = process.communicate()
```


Ther are various reasons for this. Most importantly, it `os.system` will likely not work in windows due to quoting and path conventions. Apart from that, you don't have any kind of control over the output, so it cannot be hidden in low verbose modes.

Aside from that, what was the problem with the import?